dhis-2/dhis-api/src/main/java/org/hisp/dhis/relationship/RelationshipEntity.java

```diff
@@ -45,4 +49,19 @@ public String getName()
     {
         return name;
     }
+
+    static
+    {
+        Map<String, RelationshipEntity> map = new ConcurrentHashMap<>();
+        for ( RelationshipEntity instance : RelationshipEntity.values() )
+        {
+            map.put( instance.getName(), instance );
+        }
+        ENUM_MAP = Collections.unmodifiableMap( map );
```

Comment:
I think you can use this method for ENUM_MAP and avoid the entire `static` block:

https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Maps.html#uniqueIndex-java.lang.Iterable-com.google.common.base.Function-

Reply:
Fixed using Stream.of instead of Guava